library/core/src/main/java/com/google/android/exoplayer2/text/cea/Cea608Decoder.java

```diff
@@ -284,6 +326,24 @@ protected void decode(SubtitleInputBuffer inputBuffer) {
       // If we've reached this point then there is data to process; flag that work has been done.
       captionDataProcessed = true;
 
+      if ((ccDataHeader & CC_VALID_FLAG) != CC_VALID_FLAG) {
+        // The screen needs to be cleared when the closed captions are turned off
+        // (the encoder flips the validity bit)
+        resetCueBuilders();
```

Comment:
Following on from the question of resetting when the valid flag is not set:

Isn't it a problem that above on L323, you do not reset if `ccData1 == 0 && ccData2 == 0`? If I were going to not set the valid flag, the most obvious thing I'd fill the two bytes with would be zeros. This will cause the caption to not be cleared, however you're saying that clearing is necessary.

The same applies to clearing if the parity is wrong. If that's what needs to happen, I think that decision would somehow need to be made before the `ccData1 == 0 && ccData2 == 0` case.

Reply:
Ok. Let's merge this with the order you propose. However note that we can't promise not to change it if we get reports that this is breaking streams being played by other developers / providers.